apps/site/components/EOL/EOLModal/index.tsx

```diff
@@ -0,0 +1,66 @@
+import { Modal, Title, Content } from '@node-core/ui-components/Common/Modal';
+import { useTranslations } from 'next-intl';
+import { useMemo } from 'react';
+import type { FC, ComponentProps } from 'react';
+
+import KnownSeveritySection from '#site/components/EOL/KnownSeveritySection';
+import UnknownSeveritySection from '#site/components/EOL/UnknownSeveritySection';
+import { SEVERITY_ORDER } from '#site/next.constants.mjs';
+import type { NodeRelease } from '#site/types/releases';
+import type { Vulnerability } from '#site/types/vulnerabilities';
+
+type EOLModalProps = ComponentProps<typeof Modal> & {
+  release: NodeRelease;
+  vulnerabilities: Array<Vulnerability>;
+};
+
+const EOLModal: FC<EOLModalProps> = ({
+  release,
+  vulnerabilities,
+  ...props
+}) => {
+  const t = useTranslations();
+
+  const modalHeading = release.codename
+    ? t('components.eolModal.title', {
+        version: release.major,
+        codename: release.codename,
+      })
+    : t('components.eolModal.titleWithoutCodename', { version: release.major });
+
+  useMemo(
+    () =>
+      vulnerabilities.sort(
+        (a, b) =>
+          SEVERITY_ORDER.indexOf(a.severity) -
+          SEVERITY_ORDER.indexOf(b.severity)
+      ),
+    [vulnerabilities]
```

Comment:
```suggestion
    [vulnerabilities.length]
```

Should be enough and less expensive

Reply:
this violates an eslint rule.
addressed with an elint ignore and comment within https://github.com/nodejs/nodejs.org/pull/7990/commits/37df55b63b77c158fdefe281b3cdfb3fe34bb42c